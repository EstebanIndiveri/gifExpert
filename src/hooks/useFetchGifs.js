import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs=(categorie)=>{
    const [state, setState] = useState({
        data:[],
        loading:true
    })
    useEffect(()=>{
        getGifs(categorie).then(image=>{
            setState({
                data:image,
                loading:false
            });
        })
    },[categorie])
    return state;
}