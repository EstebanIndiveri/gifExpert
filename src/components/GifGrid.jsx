import React,{ Fragment} from 'react'
import GrifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

const GifGrid = ({categorie}) => {

    const {data:images,loading}= useFetchGifs(categorie);
    
    return ( 
        <Fragment>
        <h3 className="card animate__animated animate__fadeIn">{categorie}</h3>
        {loading&&<p className="card animate__animated animate__flash">loading</p>}
        <div className="card-grid">
                {
                    images.map(image=>(
                        <GrifGridItem 
                        key={image.id}
                        {...image}
                        />
                    ))
                }
        </div>
        </Fragment>
     );
}
 
export default GifGrid;

GifGrid.propTypes={
    categorie:PropTypes.string.isRequired
}