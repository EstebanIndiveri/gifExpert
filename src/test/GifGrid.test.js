import React from 'react';
import GifGrid from '../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../hooks/useFetchGifs';
import '@testing-library/jest-dom'

jest.mock('../hooks/useFetchGifs');
describe('Si funciona <GifGrif/>', () => {
    const categorie='hola';
    test('mostrarse correctamente ', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            oading:true
        });
        const wrapper=shallow(<GifGrid categorie={categorie}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('useFetchGifs regresa items', () => {
        const gifs=[{
            id:'abc',
            url:'https://localhost',
            title:'titulo'
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            oading:true
        });
        const wrapper=shallow(<GifGrid categorie={categorie}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GrifGridItem').length).toBe(gifs.length);
    });
    
    
})
