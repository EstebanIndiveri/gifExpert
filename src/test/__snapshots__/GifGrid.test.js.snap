// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Si funciona <GifGrif/> mostrarse correctamente  1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    hola
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Si funciona <GifGrif/> useFetchGifs regresa items 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    hola
  </h3>
  <div
    className="card-grid"
  >
    <GrifGridItem
      id="abc"
      key="abc"
      title="titulo"
      url="https://localhost"
    />
  </div>
</Fragment>
`;
